crates/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -326,7 +332,7 @@ where
         nix: &NixCommandLine,
     ) -> Result<PathBuf, EnvironmentError2> {
         self.build(nix, &flox.system).await?;
```

Comment:
(non blocking) shouldn't this build be conditional?

I dont quite follow why we have activation path, which is literally just `build` and `out_link` combined.

Reply:
Probably, but that's not really the point of this PR.

Reply:
Yeah i should have noted that in the pr that introduced this, I'm just generally confused why we have `activation_path`

Reply:
That got merged before I really had a chance to even look at it

Reply:
`out_link` is to use internally as an argument to `nix --out-link`. `activation_path` is for external usage by activate. We could drop `activation_path` in favor of just `build` or something if that's clearer

Reply:
That makes sense to me